From 5b2ff215473e0526b5b24aeff4ad90d369b21c75 Mon Sep 17 00:00:00 2001
From: Julien Isorce <julien.isorce@chromium.org>
Date: Wed, 05 Feb 2020 17:59:59 +0000
Subject: [PATCH] Fix vaapi with GLX

The signature of ui's gl::GLImageGLX has changed a little bit
since "mplement GpuMemoryBuffers for EGL and GLX":
  https://chromium-review.googlesource.com/c/chromium/src/+/1984712

Bug: 1031269
Test: build with use_vaapi=true and run with --use-gl=desktop, see
Change-Id: I80b07294b9abdfa8233aaf79f7d9ec4c58117090
https: //chromium.googlesource.com/chromium/src.git/+/refs/heads/master/docs/gpu/vaapi.md#vaapi-on-linux
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2036494
Reviewed-by: Thomas Anderson <thomasanderson@chromium.org>
Reviewed-by: Miguel Casas <mcasas@chromium.org>
Commit-Queue: Julien Isorce <julien.isorce@chromium.org>
Cr-Commit-Position: refs/heads/master@{#738595}
---

diff --git a/media/gpu/vaapi/vaapi_picture_tfp.cc b/media/gpu/vaapi/vaapi_picture_tfp.cc
index 227c31b..b42620d 100644
--- media/gpu/vaapi/vaapi_picture_tfp.cc
+++ media/gpu/vaapi/vaapi_picture_tfp.cc
@@ -57,7 +57,7 @@
   if (make_context_current_cb_ && !make_context_current_cb_.Run())
     return false;

-  glx_image_ = new gl::GLImageGLX(size_, GL_RGB);
+  glx_image_ = new gl::GLImageGLX(size_, gfx::BufferFormat::BGRX_8888);
   if (!glx_image_->Initialize(x_pixmap_)) {
     // x_pixmap_ will be freed in the destructor.
     DLOG(ERROR) << "Failed creating a GLX Pixmap for TFP";
--- media/mojo/services/gpu_mojo_media_client.cc	2020-04-02 21:11:34.000000000 -0700
+++ media/mojo/services/gpu_mojo_media_client.cc	2020-04-09 00:44:58.871366432 -0700
@@ -158,6 +158,7 @@
       *d3d11_supported_configs_;

 #elif BUILDFLAG(USE_CHROMEOS_MEDIA_ACCELERATION)
+#if defined(OS_CHROMEOS)
   if (base::FeatureList::IsEnabled(kChromeosVideoDecoder)) {
     if (!cros_supported_configs_) {
       cros_supported_configs_ =
@@ -167,6 +168,7 @@
         *cros_supported_configs_;
     return supported_config_map;
   }
+#endif  //defined(OS_CHROMEOS)
 #endif

   auto& default_configs =
